diff --git a/bin/gbuild b/bin/gbuild
index 24e4e2a..77bd4a3 100755
--- a/bin/gbuild
+++ b/bin/gbuild
@@ -142,7 +142,7 @@ EOF" if build_desc["sudo"] and @options[:allow_sudo]
     build_desc["remotes"].each do |remote|
       dir = sanitize(remote["dir"], remote["dir"])
 
-      author_date = `cd inputs/#{dir} && git log --format=@%at -1 | date +"%F %T" -u -f -`.strip
+      author_date = `cd inputs/#{dir} && git log --format=@%at -1 | gdate +"%F %T" -u -f -`.strip
       raise "error looking up author date in #{dir}" unless $?.exitstatus == 0
 
       system! "copy-to-target #{@quiet_flag} inputs/#{dir} build/"
